app/screens/Lists/BooksList/index.js

```diff
@@ -7,9 +7,11 @@ import { bookData } from './constants';
 class BooksListContainer extends Component {
   renderList = ({ item }) => (
     <BookItem
+      navigation={this.props.navigation}
       image={item.image_url}
```

Comment:
this is that i said before.
You have to give the navigation behaviour to THIS smart component and pass the function reference to the children.
The the children will call the father's function passing the particular params .